ldaptive/src/test/java/org/ccci/idm/user/ldaptive/dao/LdaptiveUserDaoIT.java

```diff
@@ -406,6 +407,81 @@ public void run() {
         }
     }
 
+    /**
+     * Test findAllByQuery. we use bit math to generate and test all possible search combinations
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testFindAllByQuery() throws Exception {
+        assumeConfigured();
+        assumeGroupsConfigured();
+
+        // generate all possible intersections of query values on accounts
+        final String prefix1 = Integer.toString(RAND.nextInt(Integer.MAX_VALUE));
+        final String prefix2 = Integer.toString(RAND.nextInt(Integer.MAX_VALUE));
+        final int EMAIL = 1 << 0;
+        final int FIRST = 1 << 1;
+        final int LAST = 1 << 2;
+        final int GROUP = 1 << 3;
+        final int DEACTIVATED = 1 << 4;
+        final int MAX = 1 << 5;
+        for (int i = 0; i < MAX; i++) {
+            final User user = newUser();
+            user.setEmail(((EMAIL & i) == EMAIL ? prefix1 : prefix2) + "-" + user.getEmail());
+            user.setFirstName(((FIRST & i) == FIRST ? prefix1 : prefix2) + "-" + user.getFirstName());
+            user.setLastName(((LAST & i) == LAST ? prefix1 : prefix2) + "-" + user.getLastName());
+            user.setDeactivated((DEACTIVATED & i) == DEACTIVATED);
+            this.dao.save(user);
+
+            // add user to group1
+            if ((GROUP & i) == GROUP) {
+                this.dao.addToGroup(user, group1);
+            }
+        }
+
+        // perform all possible query combinations, we only search for prefix1.
+        for (int i = 0; i < MAX; i++) {
+            int count = MAX;
+            boolean hasFilter = false;
+            final SearchQuery query = new SearchQuery();
+            if ((EMAIL & i) == EMAIL) {
+                query.email(prefix1 + "*");
+                count /= 2;
+                hasFilter = true;
+            }
+            if ((FIRST & i) == FIRST) {
+                query.firstName(prefix1 + "*");
+                count /= 2;
+                hasFilter = true;
+            }
+            if ((LAST & i) == LAST) {
+                query.lastName(prefix1 + "*");
+                count /= 2;
+                hasFilter = true;
+            }
+            // groups aren't unique to this test, so we don't consider it as hasFilter
+            if ((GROUP & i) == GROUP) {
+                query.group(group1);
+                count /= 2;
+            }
+
```

Comment:
wasn't originally included in my search UI for The Key, and isn't one of the pre-existing `findAllBy*()` methods. I should probably add it in to `SearchQuery` since that is functionality that will be needed in the final combined admin UI